backend/src/main.js

```diff
@@ -1,114 +1,178 @@
 const express = require('express');
-const path = require('path');
 const app = express();
 const bodyParser = require('body-parser');
-const entries = require('./entries.json');
 const log = require('./libs/log.js')(module);
-const mongoose = require('./libs/mongoose');
 const passEntry = require('./models/passentry').PassEntry;
 const user = require('./models/user').user;
 const passport = require('./libs/passport.js');
-
-mongoose.initConnect();
-
-function guid() {
-    function s4() {
-        return Math.floor((1 + Math.random()) * 0x10000)
-            .toString(16)
-            .substring(1);
-    }
-    return s4() + s4() + '-' + s4() + '-' + s4() + '-' +
-        s4() + '-' + s4() + s4() + s4();
-}
+const jwt = require('jsonwebtoken');
 
 app.use(express.static('frontend/dist'));
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: true }));
+app.use(passport.initialize());
+
+
+app.get('/home', function (req, res, next) {
+	return res.redirect('/');
+});
 
 app.get('/api/entries', function (req, res, next) {
-    return passEntry.find(function(err, passEntryList){
+    return passport.authenticate('jwt', {session: 'false'}, async function (err, user, info){
         if (err){
+            res.statusCode = 401;
+            return res.send({ error: 'Server error: ' + err });
+        }
+        try {
+            const passEntryList = await passEntry.findAll({ where: {userID: user.ID} });
+            res.statusCode = 201;
+            return res.send(passEntryList);
+        } catch (err) {
             res.statusCode = 500;
             log.error('Internal error(%d): %s',res.statusCode,err.message);
             return res.send({ error: 'Server error' });
         }
-        return res.send(passEntryList);
-    });
+    })(req, res, next);
 });
 
-app.get('/api/entries/:id', function (req, res, next) {
-    passEntry.find({'id': res.params[0]}, function(err, passentry){
-        if (err){
+app.get('/api/entries/:id', async function (req, res, next) {
+    try {
+        const passEntryOne = await passEntry.findOne({ where: {ID: res.params[0]} });
```

Comment:
it is still very not right to be able to get any passentry just knowing its id. User's passentry should *only* be accessible by people who have corresponding access token.